addon-test-support/-private/weight-test-modules.js

```diff
@@ -0,0 +1,59 @@
+function getWeight(moduleName) {
+    var weight = -1;
+    if (moduleName.includes('/jshint/')) {
+        weight = 1;
+    } else if (moduleName.includes('/unit/')) {
+        weight = 10;
+    } else if (moduleName.includes('/integration/')) {
+        weight = 20;
+    } else if (moduleName.includes('/acceptance/')) {
+        weight = 150;
+    } else {
+        weight = 50;
+    }
+    return weight;
+}
+
+function maxIndex(array) {
+    return array.reduce(function(max, value, index, arr) {
+        return value > arr[max] ? index : max;
+    }, 0);
+}
+
+export default function weightTestModules(modules) {
+    var weightedModule = [];
+    var groups = [];
```

Comment:
I think we can simplify this code:
if groups was an object instead of an array, we can get rid of the weights array as such:

```
    modules.forEach(function(module) {
        const moduleWeight = getWeight(module);

        if (typeof groups[moduleWeight] == 'undefined') {
            groups[moduleWeight] = [];
        }
        groups[moduleWeight].push(module);
    });

    // returns a sort list of the weights in descending order
    const sortedWeights = Object.keys(groups).sort(function(a, b){return b-a});  
    sortedWeights.forEach( function(weight) {
        const sortedMoudleArr = groups[weight].sort();
        weightedModule = weightedModule.concat(sortedMoudleArr);
    }
     return weightedModule;
```